### PR TITLE
Fix OSB price duplication on delayed lookup

### LINKED ISSUE #5858: OSBuddy Actively traded price appears multiple time if a hiscore lookup is submitted before display
**Describe the bug**
If a hiscore lookup is initiated (say, by accidentally double-clicking a grand exchange item while searching and clicking a player's chat entry) before the "OSBuddy Actively traded price" is displayed, it will be delayed until completion of the hiscore lookup, and will display multiple times.

**To Reproduce**
1. Enter the "Choose an item" buy item search screen, search for any item
2. Quickly click on the item and on a chat line after the widget changes back to the normal chat widget
3. See error (see gif at bottom of post)

**Expected behavior**
Hiscore lookup and osbuddy price lookup are performed in parallel, and do not cause adverse effects

**Actual behavior**
OSBuddy price lookup is paused while hiscore lookup occurs, results show multiple times after the hiscore lookup completes

**Screenshots**
osbuddy actively traded price display bug

### PR DESCRIPTION
**Pull request by Karolcz125**
Fixes #5858

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java [hunk 1]
```diff
@@ -338,15 +338,14 @@ public void onGameTick(GameTick event)
...
-			}
-			catch (IOException e)
-			{
-				log.debug("Error getting price of item {}", itemId, e);
+			if (!geText.getText().contains(OSB_GE_TEXT)) {
+				try {
+					final GrandExchangeResult result = CLIENT.lookupItem(itemId);
+					final String text = geText.getText() + OSB_GE_TEXT + StackFormatter.formatNumber(result.getOverall_average());
+					geText.setText(text);
+				} catch (IOException e) {
```
Comment (forsco) [inline runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java]: new line after curly

File: runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java [hunk 2]
```diff
@@ -338,15 +338,14 @@ public void onGameTick(GameTick event)
...
-			{
-				final GrandExchangeResult result = CLIENT.lookupItem(itemId);
-				final String text = geText.getText() + OSB_GE_TEXT + StackFormatter.formatNumber(result.getOverall_average());
-				geText.setText(text);
-			}
-			catch (IOException e)
-			{
-				log.debug("Error getting price of item {}", itemId, e);
+			if (!geText.getText().contains(OSB_GE_TEXT)) {
+				try {
```
Comment (forsco) [inline runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java]: brace on new line

File: runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java [hunk 3]
```diff
@@ -338,15 +338,14 @@ public void onGameTick(GameTick event)
...
-			try
-			{
-				final GrandExchangeResult result = CLIENT.lookupItem(itemId);
-				final String text = geText.getText() + OSB_GE_TEXT + StackFormatter.formatNumber(result.getOverall_average());
-				geText.setText(text);
-			}
-			catch (IOException e)
-			{
-				log.debug("Error getting price of item {}", itemId, e);
+			if (!geText.getText().contains(OSB_GE_TEXT)) {
```
Comment (forsco) [inline runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java]: brace on new line

File: runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java [hunk 4]
```diff
@@ -338,15 +338,18 @@ public void onGameTick(GameTick event)
 
 		executorService.submit(() ->
 		{
-			try
+			if (!geText.getText().contains(OSB_GE_TEXT))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java]: you should flip this if so the diff will be small (e.g if (geText.contains)... return

**Pull request merged**